### PR TITLE
Explicitly removal of all invisible actions when a pipeline finishes.

### PR DESCRIPTION
**Pull request by drodriguezhdez**
### Requirements for Contributing to this repository

* Fill out the template below. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* The pull request must only fix one issue at the time.
* The pull request must update the test suite to demonstrate the changed functionality.
* After you create the pull request, all status checks must be pass before a maintainer reviews your contribution. For more details, please see [CONTRIBUTING](/CONTRIBUTING.md). 

### What does this PR do?

This PR removes explicitly all the `InvisibleAction` instances from the `WorkflowRun` instance used by CI Visibility. This is required to avoid memory leaks due to the action instances are not garbage collected, even if the run has finished.

As the `onFinalized` method is executed before processing the last `FlowNode`, we synchronize the clean-up using the `IsPipelineAction`.

Before the fix -> After executing 20 pipelines

image

After the fix -> After executing 20 pipelines

image

Related to #227 

### Review checklist (to be filled by reviewers)

- [ ] Feature or bug fix MUST have appropriate tests (unit, integration, etc...)
- [ ] PR title must be written as a CHANGELOG entry [(see why)](/CONTRIBUTING.md#pull-request-title)
- [ ] Files changes must correspond to the primary purpose of the PR as described in the title (small unrelated changes should have their own PR)
- [ ] PR must have one `changelog/` label attached. If applicable it should have the `backward-incompatible` label attached.
- [ ] PR should not have `do-not-merge/` label attached.
- [ ] If Applicable, issue must have `kind/` and `severity/` labels attached at least.

### CODE REVIEW

File: src/main/java/org/datadog/jenkins/plugins/datadog/listeners/DatadogBuildListener.java [hunk 1]
```diff
@@ -302,18 +309,22 @@ public void onFinalized(Run run) {
                 buildData = new BuildData(run, null);
             } catch (IOException | InterruptedException e) {
                 DatadogUtilities.severe(logger, e, "Failed to parse finalized build data");
+                cleanUpTraceActions(run);
```
Comment (AdrianLC) [inline src/main/java/org/datadog/jenkins/plugins/datadog/listeners/DatadogBuildListener.java:312]: Isn't the `finally` below called regardless of the return in this block? That would mean cleaning twice unnecessarily

File: src/main/java/org/datadog/jenkins/plugins/datadog/listeners/DatadogBuildListener.java [hunk 2]
```diff
@@ -302,18 +309,22 @@ public void onFinalized(Run run) {
...
                 return;
             }
 
             // APM Traces
             client.finishBuildTrace(buildData, run);
             logger.fine("End DatadogBuildListener#onFinalized");
+
```
Comment (AdrianLC) [inline src/main/java/org/datadog/jenkins/plugins/datadog/listeners/DatadogBuildListener.java:319]:
```suggestion
```

**Review by AdrianLC — DISMISSED** [review]
It'd be great to have a test checking this

File: src/main/java/org/datadog/jenkins/plugins/datadog/listeners/DatadogBuildListener.java (hunk 1 above)
Comment (drodriguezhdez) [reply src/main/java/org/datadog/jenkins/plugins/datadog/listeners/DatadogBuildListener.java:312]: Totally correct. I'll remove this.

**Comment by drodriguezhdez** [comment]
> It'd be great to have a test checking this

I added assertions for this behavior.

**Review by FlorianVeaux — APPROVED** [review]

**Pull request merged**